### DISCUSSION TITLE
Will the storage on my Seedvault flash drive run out?

### DISCUSSION THREAD

**Discussion by lazytownfan** [seedvault-app/seedvault#337] [Q&A] [answered]
I would like to know what happens to the flash drive I use for Seedvault when the flash drive I'm using runs out of storage space.

Will I have to get a new flash drive for my Seedvault backups? Or, is Seedvault somehow capable of deleting the oldest data in order to make room for new Seedvault backups?

**Comment by grote** [comment, marked as answer]
There's no automatic storage handling implemented for app backups so far. "File/storage backups" do delete all backups automatically.

For app backups, you should have numeric folders inside `.SeedvaultAndroidBackup` which you can delete if they haven't been changed recently and you have other backups available.

**Reply by lazytownfan** [reply]
Does this still apply even when the flash drive is only used for Seedvault and nothing else?

I don't store any other data on my Seedvault flash drive.

**Reply by grote** [reply]
Yes, Seedvault doesn't care about other data. It only touches what's inside that `.SeedvaultAndroidBackup` folder.

**Comment by LeSpocky** [comment]
Is this still recent behaviour?  I'm using a Fairphone 3 with LineageOS 20 (Android 13) for two weeks now and Seedvault already uses 100 GB of the 128 GB uSD card (using the external sd as backup target).  One backup including apps is around 3.3 GB and there are ~30 backups stored.  Is it possible to limit the number of backups kept by Seedvault?

**Reply by steadfasterX** [reply]
see: https://github.com/seedvault-app/seedvault/issues/559 :)

**Comment by milindpatel63** [comment]
> There's no automatic storage handling implemented for app backups so far. "File/storage backups" do delete all backups automatically.
> 
> For app backups, you should have numeric folders inside `.SeedvaultAndroidBackup` which you can delete if they haven't been changed recently and you have other backups available.

If those numeric folders haven't been changed recently, and I delete them, then a backup of an app gets deleted right? But how do I know if there's a newer backup of that app is available?

Does seedvault make new numbered folders each time it backups for each app?

Also, what about Storage backup? If I remove a folder from backup which was previously getting backed up, will it also remove the backup for that folder when the backup runs next? Or will it be permanently saved in my backup forever?

Would really love if there were some docs and faq regarding this.

**Reply by grote** [reply]
> Does seedvault make new numbered folders each time it backups for each app?

No, it creates a new "restore set" only under specific circumstances like after selecting a new backup location.

> Would really love if there were some docs and faq regarding this.

Would you write one based on the answers you get here? However, these things are considered internal implementation details and may change in any new version.

I personally think a feature like https://github.com/seedvault-app/seedvault/issues/100 would make knowing those internal details mostly unnecessary.

**Reply by o-alquimista** [reply]
> Also, what about Storage backup? If I remove a folder from backup which was previously getting backed up, will it also remove the backup for that folder when the backup runs next? Or will it be permanently saved in my backup forever?

@grote I'd like to know this too, please. I see that #559 was implemented, but it's not clear if it applies to the storage backup (user folder backups).

**Reply by grote** [reply]
Storage backups also auto-delete old backups with a reasonable default. You should never manually remove anything in your backup directory.
